app/src/main/java/com/epfl/drawyourpath/database/FirebaseDatabaseUtils.kt

```diff
@@ -80,27 +82,36 @@ object FirebaseDatabaseUtils {
      * @return the run corresponding to the data
      */
     fun transformRun(data: DataSnapshot?): Run? {
-        // TODO: THis function must be refectored with the new strcuture of a run and a path
-        /*
-        val points = data?.child("path")?.child("points")?.children?.mapNotNull {
-            val lat = getNumber(it.child("latitude"))?.toDouble()
-            val lon = getNumber(it.child("longitude"))?.toDouble()
-            if (lat != null && lon != null) {
-                LatLng(lat, lon)
-            } else {
-                Log.w(this::class.java.name, "A coordinate was badly formatted.")
-                null
+        val sections = data?.child("path")?.child("points")?.children?.mapNotNull { section ->
+            section.children.mapNotNull { point ->
+                val lat = getNumber(point.child("latitude"))?.toDouble()
+                val lon = getNumber(point.child("longitude"))?.toDouble()
+                if (lat != null && lon != null) {
+                    LatLng(lat, lon)
+                } else {
+                    Log.w(this::class.java.name, "A coordinate was badly formatted.")
+                    null
+                }
             }
         } ?: emptyList()
```

Comment:
Map returns an empty list so I don't think we need to filter `null` values.

```suggestion
        val sections = data?.child("path")?.child("points")?.children?.map { section ->
            section.children.mapNotNull { point ->
                val lat = getNumber(point.child("latitude"))?.toDouble()
                val lon = getNumber(point.child("longitude"))?.toDouble()
                if (lat != null && lon != null) {
                    LatLng(lat, lon)
                } else {
                    Log.w(this::class.java.name, "A coordinate was badly formatted.")
                    null
                }
            }
        } ?: emptyList()
```